client/app/stores/message_store.coffee

```diff
@@ -606,13 +606,18 @@ class MessageStore extends Store
         if param.type is 'message' and not _conversationMemoize.size
             messages = _currentMessages
 
-        getMessage = ->
-            index0 = messages.toArray().findIndex (message) ->
-                if param.conversationID isnt 'undefined'
-                    return param.conversationID is message.get 'conversationID'
-                return messageID is message.get 'id'
-            index = param.transform index0
-            messages.get index
+        getMessage = =>
+            _newMessage = (index) ->
```

Comment:
That one is hard to understand. Moreover there is a return true without return false in other case. I think comments, explicitly handle all cases and better variable names will improve it.